### DISCUSSION TITLE
@xstate/test starting from the beginning everytime

### DISCUSSION THREAD

**Discussion by serranomorante** [statelyai/xstate#4042] [General]
Maybe I don't understand Model-based testing with @xstate/test but is this sequence the desired result when you have [this](https://stackblitz.com/edit/stackblitz-starters-g2mrck?file=sample.test.js) machine? (execute `npm run test` on terminal)

```bash
 PASS  ./sample.test.js
  createEntity
    reaches state: "#createEntity.idle" 
      ✓ via  (49 ms)
    reaches state: "#createEntity.fillBasicDetails" 
      ✓ via FILL_BASIC_DETAILS (48 ms)
    reaches state: "#createEntity.fillTriggerData" 
      ✓ via FILL_BASIC_DETAILS → FILL_TRIGGER_DATA (49 ms)
    reaches state: "#createEntity.fillStepsData" 
      ✓ via FILL_BASIC_DETAILS → FILL_TRIGGER_DATA → FILL_STEPS_DATA (48 ms)
    reaches state: "#createEntity.fillUnsubscribeConditions" 
      ✓ via FILL_BASIC_DETAILS → FILL_TRIGGER_DATA → FILL_STEPS_DATA → FILL_UNSUBSCRIBE_CONDITIONS (50 ms)

Test Suites: 1 passed, 1 total
Tests:       5 passed, 5 total
Snapshots:   0 total
Time:        0.888 s, estimated 1 s
Ran all test suites.
```
If I use that machine with Cypress, each test starts from the beginning (from FILL_BASIC_DETAILS event), I don't want that... I was expecting to go from start to end in just one sequence.

Is there any advise on how to make the @xstate/test a little bit more predictable while I'm trying to develop my tests?

**Comment by davidkpiano** [comment]
In the next major version of `@xstate/test`, paths are deduplicated as best as possible.

See for yourself in `@xstate/test@alpha` - there will be only one test:

https://stackblitz.com/edit/stackblitz-starters-nispqp?file=sample.test.js


<img width="722" alt="CleanShot 2023-05-29 at 20 16 28@2x" src="https://github.com/statelyai/xstate/assets/1093738/7771400e-9de0-4064-b5bc-77233e872be4">
